worlds/sc2/mission_order/layout_types.py

```diff
@@ -138,8 +138,20 @@ def get_grid_dimensions(size: int) -> Tuple[int, int, int]:
         best_dimension = min(dimension_candidates, key=sum)
         return best_dimension
 
+    @staticmethod
+    def manhattan_distance(point1: Tuple[int, int], point2: Tuple[int, int]) -> int:
+        return abs(point1[0] - point2[0]) + abs(point1[1] - point2[1])
+    
+    @staticmethod
+    def euclidean_distance_squared(point1: Tuple[int, int], point2: Tuple[int, int]) -> int:
+        return (point1[0] - point2[0]) ** 2 + (point1[1] - point2[1]) ** 2
+    
+    @staticmethod
+    def euclidean_distance(point1: Tuple[int, int], point2: Tuple[int, int]) -> float:
+        return math.sqrt(Grid.euclidean_distance_squared(point1, point2))
+
     def get_grid_coordinates(self, idx: int) -> Tuple[int, int]:
-        return math.floor(idx / self.width), (idx % self.width)
+        return (idx % self.width), math.floor(idx / self.width)
```

Comment:
Can use `x // y` instead of `math.flooar(x/y)` for integer division, which should give the same result for positive indices.